src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -973,8 +973,8 @@ public function __construct()
 EOF;
         }
         if ($this->asFiles) {
-            $code = str_replace('$parameters', "\$buildParameters;\n    private \$parameters", $code);
-            $code = str_replace('__construct()', '__construct(array $buildParameters = array())', $code);
+            $code = str_replace('$parameters', "\$buildParameters;\n    private \$containerDir;\n    private \$parameters", $code);
+            $code = str_replace('__construct()', '__construct(array $buildParameters = array(), $dir = __DIR__)', $code);
```

Comment:
the default value of the argument is broken. The compiled code expects the `containerDir` to end with a directory separator.

Reply:
fixed